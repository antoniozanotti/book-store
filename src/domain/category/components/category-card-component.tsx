import { UiTitleComponent } from "@/domain/ui/components/ui-title-component";
import { CategoryWithPagePrisma } from "../types/category-prisma";
import { CategoryLinkComponent } from "./category-link-component";
import { UiFigureComponent } from "@/ui-figure-component";

export function CategoryCardComponent({
  category,
}: {
  category: CategoryWithPagePrisma;
}) {
  return (
    <article className="snap-start p-5 w-[227px] h-[361px] flex-none text-sm text-pretty">
      <CategoryLinkComponent category={category}>
        <UiFigureComponent
          src={`category/${category.page.slug}`}
          width={187}
          height={269}
          alt={category.name}
          className="w-[187px] h-[249px] overflow-hidden"
          innerImgStyle="hover:-mt-[20px]"
        />
      </CategoryLinkComponent>
      <UiTitleComponent level="p" size="small" className="line-clamp-2 mt-2">
        <CategoryLinkComponent category={category}>
          {category.name}
        </CategoryLinkComponent>
      </UiTitleComponent>
    </article>
  );
}
