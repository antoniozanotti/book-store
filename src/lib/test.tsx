import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
