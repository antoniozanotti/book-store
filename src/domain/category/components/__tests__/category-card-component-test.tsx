import { render, screen } from "@/lib/test";
import { CategoryCardComponent } from "../category-card-component";
import { categoriesWithPageMock } from "../../__tests__/categories-with-page-mock";

describe("CategoryCardComponent", () => {
  it("should render", async () => {
    const { container } = render(
      <CategoryCardComponent category={categoriesWithPageMock[0]} />
    );
    expect(container).toMatchSnapshot();
  });
});
