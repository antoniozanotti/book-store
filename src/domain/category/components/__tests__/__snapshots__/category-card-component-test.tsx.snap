// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCardComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <article
      class="snap-start p-5 w-[227px] h-[361px] flex-none text-sm text-pretty"
    >
      <a
        href="/category/arts-and-entertainment"
        title="Arts and Entertainment"
      >
        <figure
          class="bg-green-300 w-[187px] h-[249px] overflow-hidden"
        >
          <img
            alt="Arts and Entertainment"
            class="transition-all hover:-mt-[20px]"
            data-nimg="1"
            decoding="async"
            height="269"
            loading="lazy"
            sizes="187px"
            src="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_5523,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0"
            srcset="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_16,h_23,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 16w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_32,h_46,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 32w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_48,h_69,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 48w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_64,h_92,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 64w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_96,h_138,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 96w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_128,h_184,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 128w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_256,h_368,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 256w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_384,h_552,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 384w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_640,h_920,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 640w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_750,h_1078,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 750w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_828,h_1191,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 828w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1080,h_1553,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 1080w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1200,h_1726,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 1200w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1920,h_2761,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 1920w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_2048,h_2946,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 2048w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_5523,g_auto/f_auto/q_auto/v1/category/arts-and-entertainment?_a=BAVARSDW0 3840w"
            style="color: transparent;"
            width="187"
          />
        </figure>
      </a>
      <p
        class="font-primary text-pretty text-2xl line-clamp-2 mt-2"
      >
        <a
          href="/category/arts-and-entertainment"
          title="Arts and Entertainment"
        >
          Arts and Entertainment
        </a>
      </p>
    </article>
  </div>
</div>
`;
